api/database/migrations/2024_05_17_184018_update_notifications_store_enabled.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+use App\Enums\NotificationFamily;
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    private $notificationFamilies = [
+        NotificationFamily::APPLICATION_UPDATE->name,
+        NotificationFamily::JOB_ALERT->name,
+    ];
+
+    private function invertFamilies(?string $row): string
+    {
+        $json = json_decode($row ?? '[]');
+        $value = [];
+
+        foreach ($json as $family) {
+            array_push($value, $family);
+        }
+
+        $inverted = array_diff($this->notificationFamilies, $value);
+
+        return json_encode(array_values($inverted));
+    }
+
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+
+        Schema::table('users', function (Blueprint $table) {
+            $table->jsonb('enabled_email_notifications')->nullable();
+            $table->jsonb('enabled_in_app_notifications')->nullable();
+        });
+
+        $rows = DB::table('users')
+            ->select(DB::raw('ignored_email_notifications, ignored_in_app_notifications, id'))
+            ->get();
+
+        $rows->each(function ($row) {
```

Comment:
Oh no, not the imperative rows->each again.  :laughing:

I'm worried about this taking a while during deployment and the possibility of changes during the updates since it isn't atomic.  Since we have only two real notification families right now I wonder about doing something like this:
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/8978655/31472b05-ba85-4c6e-b01b-56ab9ad4e42f)

Reply:
Thanks for your help with this. I think we got it working in [update migration to avoid loop](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10430/commits/6d13fb3a10412bf3528cebc62f85649561b71d46)